### PR TITLE
[no-master] Enable Scala 2.13.0 in the tools and all the other artifacts.

### PR DESCRIPTION
**Pull request by sjrd**
It's [no-master]  because

* The `@deprecatedName` and `scopt` commits are irrelevant in master.
* The other two commits are virtually all-conflicts if I try to cherry-pick them on master. An independent forward-port will be easier to review.

### CODE REVIEW

File: tools/shared/src/main/scala/org/scalajs/core/tools/linker/CollectionsCompat.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+import scala.collection.mutable
+
+private[linker] object CollectionsCompat {
+  implicit class MutableMapCompatOps[K, V](val __self: mutable.Map[K, V])
+      extends AnyVal {
+
+    // filterInPlace replaces retain
+    def filterInPlace(p: (K, V) => Boolean): Unit = {
+      // Believe it or not, this is the implementation of `retain` in 2.12.x:
```
Comment (gzm0) [inline tools/shared/src/main/scala/org/scalajs/core/tools/linker/CollectionsCompat.scala]: Hahahahaha, were you anticipating me complaining :P

File: tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/optimizer/GenIncOptimizer.scala [hunk 2]
```diff
@@ -1027,31 +1029,37 @@ object GenIncOptimizer {
...
+    type ParIterable[V] <: AnyRef
+    type Addable[V] <: AnyRef
 
     def emptyAccMap[K, V]: AccMap[K, V]
     def emptyMap[K, V]: Map[K, V]
     def emptyParMap[K, V]: ParMap[K, V]
     def emptyParIterable[V]: ParIterable[V]
 
     // Operations on ParMap
+    def parMapIsEmpty[K, V](map: ParMap[K, V]): Boolean
```
Comment (gzm0) [inline tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/optimizer/GenIncOptimizer.scala]: Simply name this `isEmpty`? `parMap` seems superfluous (I understand that overloading is a problem with types that end up being erased, but the whole thing is already terribly inconsistent anyways :S).

File: tools/shared/src/main/scala/org/scalajs/core/tools/linker/CollectionsCompat.scala (hunk 1 above)
Comment (sjrd) [reply tools/shared/src/main/scala/org/scalajs/core/tools/linker/CollectionsCompat.scala]: Actually I was anticipating myself looking back at this code and being appalled at the inefficiency of it.

**Comment by sjrd** [comment]
Updated.

Besides addressing the review, I have also added the whitelist and blacklist for the `scalaTestSuite`, so that the CI can pass. This discovered two bugs filed as #3704 and #3705, which should be fixed separately, as followups.

File: scala-test-suite/src/test/resources/2.13.0/BlacklistedTests.txt [hunk 3]
```diff
@@ -122,12 +172,20 @@ scala/util/matching/RegexTest.scala
 # Require strict-floats
 scala/math/BigDecimalTest.scala
 
-# Difference of getClass() on primitive values
-scala/collection/immutable/RangeTest.scala
-
-# Bugs
-scala/collection/convert/MapWrapperTest.scala
+# Fails for a BigDecimal range with augmented precision (might be an actual bug)
```
Comment (gzm0) [inline scala-test-suite/src/test/resources/2.13.0/BlacklistedTests.txt]: Do we need to file an issue for this at least?

Comment (sjrd) [reply scala-test-suite/src/test/resources/2.13.0/BlacklistedTests.txt]: Filed as #3706.

**Pull request merged**